### PR TITLE
Use assetsPath as Webpack publicPath

### PR DESCRIPTION
**Pull request by dallonf**
## Changes/Tasks

- [x] Changed `output.publicPath` to the `REACT_STATIC_ASSETS_PATH ` variable

## Motivation and Context

Based on what `publicPath` [does in Webpack](https://webpack.js.org/configuration/output/#outputpublicpath), I believe it's actually better set to what React-Static calls `assetsPath`. The dev Webpack configuration already works this way.

In our use case, the assets are hosted on a different CDN than the actual site - if the `siteRoot` is `https://www.example.com`, the `assetRoot` is `https://xxxxx.cloudfront.net`. However, certain imports (mostly images imported like `import imgSrc from '../img.png';`) were still resulting in a path like `https://www.example.com/static/img.png`.

## Types of changes

- [ ] Refactoring/add tests (refactoring or adding test which isn't a fix or add a feature)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have updated the documentation accordingly*
- [x] I have updated the CHANGELOG with a summary of my changes
- [ ] My changes have tests around them**

*This is just a bugfix, I don't think documentation changes are necessary
** Don't think there's anything to test here

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@
 - Fix missing state.siteData in dev ([#1148](https://github.com/react-static/react-static/pull/1148))
 - Add clickable dev-server url ([#1306](https://github.com/react-static/react-static/pull/1306))
 
+### Fixes & Optimizations
```
Comment (SleeplessByte) [inline CHANGELOG.md]: This would fall under `improved`

**Review by SleeplessByte — APPROVED** [review]
Can you just remove the heading?

**Comment by SleeplessByte** [comment]
Perfect!

cc @nddery @jviotti 

If you have a spare few minutes, would this change work for your set-ups?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 - Fix "can not read property `catch` of `undefined`" ([#1313](https://github.com/react-static/react-static/pull/1313))
 - Fix missing state.siteData in dev ([#1148](https://github.com/react-static/react-static/pull/1148))
 - Add clickable dev-server url ([#1306](https://github.com/react-static/react-static/pull/1306))
+- Fix `output.publicPath` in dev build when `assetPath` is set ([#1329](https://github.com/react-static/react-static/pull/1329))
```
Comment (nddery) [inline CHANGELOG.md]: Shouldn't this be "in prod build" ?

**Review by nddery — COMMENTED** [review]
I'm thinking this might lead to issues when `config.assetsPath` is not absolute (ie.: when it's set to "assets", for example) and when the user sets `config.basePath`. Because of https://github.com/react-static/react-static/blob/master/packages/react-static/src/static/getConfig.js#L141-L143 the asset path would be wrong no ?

This also seems to be breaking when only a `basePath` is set, because assets being prefetched do not include the base path.

File: CHANGELOG.md (hunk 2 above)
Comment (dallonf) [reply CHANGELOG.md]: Yes... yes it should. Oops

**Comment by dallonf** [comment]
> I'm thinking this might lead to issues when `config.assetsPath` is not absolute (ie.: when it's set to "assets", for example) and when the user sets `config.basePath`. Because of https://github.com/react-static/react-static/blob/master/packages/react-static/src/static/getConfig.js#L141-L143 the asset path would be wrong no ?
> 
> This also seems to be breaking when only a `basePath` is set, because assets being prefetched do not include the base path.

This sounds like the issue is a lot more complicated than anticipated. I'm not sure of what the real solution is. Should I close this PR and open a bug ticket instead describing the problem?

**Comment by SleeplessByte** [comment]
Most likely it "just" needs a check to see if those bases are set, and if not, "fix" the path resolution. There are various bugs currently due to switching to absolute paths and I am looking into "undoing it", by means of supporting absolute paths only in specific places.

Anyway, both @nddery and @jviotti have build systems currently where they build in a different "tree" than their project lives (if I recall correctly), and they've both submitted various PRs to make that work. If your changes work with their systems, and we can capture edge cases, this is fine to merge. Will be changed in RS 8, which is weeks to months out.

**Comment by nddery** [comment]
We've configured our react-static to build in the default `dist` folder. We do however use the `basePath` configuration, as well as `assetsPath` (set to "assets" - same value as `paths.assets`). Being able to set an `assetsPath` that points to a CDN would, I think, be useful for us, too. 

The absolute paths issue is a blocker for us to upgrade to v7.2.2, which I've been trying to address in https://github.com/react-static/react-static/issues/1303 (but have not been able to put a lot of time into this lately...). I'd be interested to hear more about your thoughts on "supporting absolute paths only in specific places" (which might be a different topic than this PR)!

My biggest hurdle is the shear number of different configuration possible, which I find to be the same issue here, a simple change can have big impact on different configurations. I've been thinking that a thorough review of every configuration options and how they interplay with each other might help.

**Comment by SleeplessByte** [comment]
> My biggest hurdle is the shear number of different configuration possible, which I find to be the same issue here, a simple change can have big impact on different configurations. I've been thinking that a thorough review of every configuration options and how they interplay with each other might help.

Perhaps we can slash a few?

**Comment by kvet** [comment]
@SleeplessByte will this ever be released? We are patching `react-static` in order to get it working

**Comment by SleeplessByte** [comment]
I don't think it's currently finished

**Pull request closed without merging**